File: lab3/Java/src/dataAccess/repository/IBookRepository.java
package dataAccess.repository;

import dataAccess.dbmodel.BookDto;

import java.util.List;

/**
 * Created by Alex on 07/03/2017.
 * Adapted by Andrei 16/06/2018.
 */
public interface IBookRepository {

    List<BookDto> findAll();

    BookDto findById(Long id);

    boolean create(BookDto book);

    void update(BookDto book);
}

File: lab4/LayersWithDependencyInjection/DataAccess.Contracts/Services/IBookRepository.cs
using DataAccess.Contracts.Models;

namespace DataAccess.Contracts.Services
{
    public interface IBookRepository
    {
        BookDto GetById(int id);
        BookDto[] GetAll();

        bool Add(BookDto book);
        void Update(BookDto book);

        void Delete(int id);
    }
}
